lib/shoulda/matchers/active_record/association_matcher.rb

```diff
@@ -1586,7 +1692,8 @@ def matches?(subject)
             touch_correct? &&
             types_correct? &&
             strict_loading_correct? &&
-            submatchers_match?
+            submatchers_match? &&
+            deprecated_correct?
```

Comment:
Let's leave the `submatchers_match?` as the last check, as they're usually a slower check.